src/System.Net.Http/src/System/Net/HttpClient.cs

```diff
@@ -156,41 +156,124 @@ public Task<Stream> GetStreamAsync(string requestUri)
         // Unbuffered by default
         public Task<Stream> GetStreamAsync(Uri requestUri)
         {
-            return GetContentAsync(requestUri, HttpCompletionOption.ResponseHeadersRead, Stream.Null,
+            return GetStreamContentAsync(requestUri, HttpCompletionOption.ResponseHeadersRead,
                 content => content.ReadAsStreamAsync());
         }
 
-        private Task<T> GetContentAsync<T>(Uri requestUri, HttpCompletionOption completionOption, T defaultValue,
-            Func<HttpContent, Task<T>> readAs)
+        private Task<string> GetStringContentAsync(Uri requestUri, HttpCompletionOption completionOption, 
+            Func<HttpContent, Task<string>> readAs)
         {
-            TaskCompletionSource<T> tcs = new TaskCompletionSource<T>();
+            var tcs = new TaskCompletionSource<string>(readAs);
```

Comment:
(I'll refrain from reviewing the rest of the PR until that's commented on, as is it appears most of the change relies on passing state through in that fashion.  If the team is ok with it, I am, too.)